### PR TITLE
Cosmos-level Vbank governance

### PR DESCRIPTION
**Pull request by michaelfig**
This PR is to get our toes wet with Cosmos-level governance.  The intention is to allow `x/swingset` to govern cosmic-swingset scheduler parameters (like number of computrons per block).

To fix some brokenness caused by RESM->NESM, we convert `calc-gci.js` and `add-gci-ingress.js` to NESM, and move `check-validator.js` to have a `.cjs` extension.

It also upgrades to the latest agoric-labs/cosmos-sdk (v0.43.0-alpha.agoric).

And just for good measure, we implement `ag-cosmos-helper query vbank params`.

### CODE REVIEW

File: golang/cosmos/x/vbank/client/rest/query.go [hunk 1]
```diff
@@ -0,0 +1,309 @@
...
+
+	"github.com/cosmos/cosmos-sdk/x/distribution/client/common"
+	"github.com/cosmos/cosmos-sdk/x/distribution/types"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/rest"
+)
+
+func registerQueryRoutes(clientCtx client.Context, r *mux.Router) {
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/client/rest/query.go]: Are we intentionally squatting on the namespace normally used by `x/distribution`? If so, please write a prominent comment explaining the intention.

File: golang/cosmos/x/vbank/client/rest/tx.go [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+		WithdrawAddress sdk.AccAddress `json:"withdraw_address" yaml:"withdraw_address"`
+	}
+
+	fundCommunityPoolReq struct {
+		BaseReq rest.BaseReq `json:"base_req" yaml:"base_req"`
+		Amount  sdk.Coins    `json:"amount" yaml:"amount"`
+	}
+)
+
+func registerTxHandlers(clientCtx client.Context, r *mux.Router) {
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/client/rest/tx.go]: Here also, if the squatting on `/distribution/...` is intentional, document it.

File: golang/cosmos/x/vbank/client/rest/tx.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+	// Withdraw all delegator rewards
+	r.HandleFunc(
+		"/distribution/delegators/{delegatorAddr}/rewards",
+		newWithdrawDelegatorRewardsHandlerFn(clientCtx),
+	).Methods("POST")
+
+	// Withdraw delegation rewards
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/client/rest/tx.go]: ...for a single validator.

File: golang/cosmos/x/vbank/client/rest/rest.go [hunk 4]
```diff
@@ -0,0 +1,57 @@
+package rest
```
Comment (JimLarson) [inline golang/cosmos/x/vbank/client/rest/rest.go]: Consider writing tests for this package, or make a TODO/issue to do so later.

**Comment by JimLarson** [comment]
Fixes #3424

**Review by JimLarson — APPROVED** [review]
LGTM with suggested cleanups.

File: golang/cosmos/x/vbank/client/rest/rest.go (hunk 4 above)
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/rest.go]: I tried to port the tests from Cosmos-SDK, but all the client tests are intertwined with the "simapp", which is incompatible with externally-defined modules like what we have.

So yes, yuck.

File: golang/cosmos/x/vbank/client/rest/query.go [hunk 5]
```diff
@@ -0,0 +1,309 @@
...
+		if !ok {
+			return
+		}
+
+		clientCtx = clientCtx.WithHeight(height)
+		rest.PostProcessResponse(w, clientCtx, res)
+	}
+}
+
+// HTTP request handler to query a delegation rewards
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/query.go]: Accidental.  Deleted.

File: golang/cosmos/x/vbank/client/rest/query.go [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+		delegatorAddr, ok := checkDelegatorAddressVar(w, r)
+		if !ok {
+			return
+		}
+
+		params := types.NewQueryDelegatorParams(delegatorAddr)
+		bz, err := clientCtx.LegacyAmino.MarshalJSON(params)
+		if err != nil {
+			rest.WriteErrorResponse(w, http.StatusBadRequest, fmt.Sprintf("failed to marshal params: %s", err))
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/query.go]: Agreed.  Fortunately, I could delete this code.

File: golang/cosmos/x/vbank/client/rest/query.go [hunk 7]
```diff
@@ -0,0 +1,309 @@
...
+		if !ok {
+			return
+		}
+
+		clientCtx, ok = rest.ParseQueryHeightOrReturnBadRequest(w, clientCtx, r)
+		if !ok {
+			return
+		}
+
+		bin := clientCtx.LegacyAmino.MustMarshalJSON(types.NewQueryValidatorOutstandingRewardsParams(validatorAddr))
```
[the remark this answers is not in the document]
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/query.go]: Deleted.

File: golang/cosmos/x/vbank/client/rest/query.go (hunk 1 above)
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/query.go]: You found the problem!  I accidentally copied and committed code that wasn't needed.

File: golang/cosmos/x/vbank/client/rest/tx.go (hunk 2 above)
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/tx.go]: Deleted.

File: golang/cosmos/x/vbank/client/rest/rest.go (hunk 4 above)
Comment (michaelfig) [reply golang/cosmos/x/vbank/client/rest/rest.go]: Deleted.

**Pull request merged**